gwin/calibration.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2018 Colm Talbot
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+""" Functions for adding calibration factors to waveform templates.
+"""
+
+import numpy as np
+from scipy.interpolate import UnivariateSpline
+from pycbc.types import FrequencySeries
+import logging
+
+
+class Recalibrate(object):
+
+    name = 'none'
+
+    def __init__(self, ifo_name):
+        self.ifo_name = ifo_name
+        self.params = dict()
+
+    def apply_calibration(self, strain):
+        """Apply calibration model
+
+        This method should be overwritten by subclasses
```

Comment:
Use abstract base classes here. See, e.g., `BaseModel` in models/base.py. To use:
 1. `from abc import (ABCMeta, abstractmethod)`.
 2. Set `__metaclass__ = ABCMeta` in your base class.
 3. Stick the `@abstractmethod` decorator on any method you want inheriting classes to implement. If a class that inherits from the base class does not implement the method will raise an error if you try to initialize it.